WinCompatibilityPack/WinCompatibilityPack.psm1

```diff
@@ -0,0 +1,804 @@
+#>
+function Initialize-WinSession
+{
+    [CmdletBinding()]
+    [Alias("iwins")]
+    [OutputType("System.Management.Automation.Runspaces.PSSession")]
+    Param (
+
+        # If you don't want to use the default compatibility session, use
+        # this parameter to specify the name of the computer on which to create
+        # the compatibility session.
+        # (Defaults to 'localhost')
+        [Parameter(Mandatory=$false,Position=0)]
+        [String]
+        [Alias("Cn")]
+            $ComputerName = $script:DefaultComputerName,
+
+        # Specifies the configuration to connect to when creating the compatibility session
+        # (Defaults to 'Microsoft.PowerShell')
+        [Parameter()]
+        [String]
+            $ConfigurationName = $script:DefaultConfigurationName,
+
+        # The credential to use when connecting to the target machine/configuration
+        [Parameter()]
+        [PSCredential]
+            $Credential,
+
+        # If present, the specified session object will be returned
+        [Parameter()]
+        [Switch]
+            $PassThru,
+
+        # Instead of initializing the session, show what actions will be taken instead.
+        [Parameter()]
+        [Switch]
+            $WhatIf
+    )
+
+    [bool] $verboseFlag = $PSBoundParameters['Verbose']
+
+    if ($ComputerName -eq ".")
+    {
+        $ComputerName = "localhost"
+    }
+
+    Write-Verbose -Verbose:$verboseFlag "Initializing the compatibility session on host '$ComputerName'."
+    if ($Credential)
+    {
+        $script:SessionName = "win-$($Credential.UserName)"
+    }
+    else
+    {
+        $script:SessionName = "win-$((Get-ChildItem env:UserName).Value)"
+    }
+    Write-Verbose -Verbose:$verboseFlag "The compatibility session name is '$script:SessionName'."
+
+    # BUGBUG - need to deal with the possibilities of multiple sessions
+    $session = Get-PSSession | Where-Object {
+        $_.ComputerName      -eq $ComputerName -and
+        $_.ConfigurationName -eq $ConfigurationName -and
+        $_.Name              -eq $script:SessionName
+    }
+
+    if ($session -and $session.State -ne "Opened")
+    {
+        Write-Verbose -Verbose:$verboseFlag "Removing closed compatibility session."
+        Remove-PSSession $session
+        $session = $null
+    }
+
+    if (! $session)
+    {
+        $SessionOpts = @{}
+        if ($Credential)
+        {
+            $SessionOpts.Credential = $Credential
+        }
+        if ($ComputerName -eq "." -or $ComputerName -eq (hostname) -or $ComputerName -eq "localhost")
+        {
+            $SessionOpts.EnableNetworkAccess = $true
+            
+        }
+        Write-Verbose -Verbose:$verboseFlag "Creating a new compatibility session."
+        ##BUGBUG need to deal with the case where there might be multiple sessions because someone hit ctrl-C
+        $session = New-PSSession -Verbose:$verboseFlag -Name $script:sessionName -ComputerName $ComputerName `
```

Comment:
Please use splatting instead of backticks.

Reply:
Sure.